packages/core/src/request.ts

```diff
@@ -31,19 +40,20 @@ function enhanceEventWithSdkInfo(event: Event, sdkInfo?: SdkInfo): Event {
 /** Creates a SentryRequest from a Session. */
 export function sessionToSentryRequest(session: Session | SessionAggregates, api: APIDetails): SentryRequest {
   const sdkInfo = getSdkMetadataForEnvelopeHeader(api);
-  const envelopeHeaders = JSON.stringify({
+  const envelopeHeaders = {
     sent_at: new Date().toISOString(),
     ...(sdkInfo && { sdk: sdkInfo }),
     ...(!!api.tunnel && { dsn: dsnToString(api.dsn) }),
-  });
-  // I know this is hacky but we don't want to add `session` to request type since it's never rate limited
-  const type: SentryRequestType = 'aggregates' in session ? ('sessions' as SentryRequestType) : 'session';
-  const itemHeaders = JSON.stringify({
-    type,
-  });
+  };
+
+  // I know this is hacky but we don't want to add `sessions` to request type since it's never rate limited
+  const type = 'aggregates' in session ? ('sessions' as SentryRequestType) : 'session';
 
+  // Have to cast type because envelope items do not accept a `SentryRequestType`
```

Comment:
Any reason to not just fix this so they would accept one? (I'll admit I also don't totally get the "no rate limiting means we shouldn't include `sessions` in the type" logic above. I know that it didn't originate in this PR, but it feels relevant to this question.)

Reply:
We should get to this when we examine the transports API in general - for the time being I'll leave it though.

Reply:
Okay. Let's add a flag so we remember to come back to it.

```suggestion
  // TODO (v7) Have to cast type because envelope items do not accept a `SentryRequestType`
```